src/nft-card.ts

```diff
@@ -50,123 +71,101 @@ export class NftCard extends LitElement {
         transform: rotateY(180deg);
       }
     `
-  }
-
-  /* User configurable properties */
-  @property({type: Boolean}) public horizontal = false
-  @property({type: String})  public contractAddress
-  @property({type: String})  public tokenId
-  @property({type: String})  public width = this.horizontal ? '670px' : '388px'
-  @property({type: String})  public height = this.horizontal ? '250px' : '560px'
-  @property({type: String})  public network = 'mainnet'
-
-  @property({type: String})  private flippedCard = false
-  @property({type: Boolean}) private loading = true
-
-  @property({type: Object})  public asset
-  @property({type: Object})  public traitData = {}
-  @property({type: String})  public account
-
-  private getNetwork() {
-    switch (this.network) {
-      case 'mainnet': return Network.Main
-      case 'rinkeby': return Network.Rinkeby
     }
-  }
-
-  constructor() {
-    super()
-  }
-
-  /**
-   * ConnectedCallback - Invoked when a component is added to the document’s DOM.
-   * Grabs data from the OpenSea SDK and populates data objects to be passed to
-   * child components.
-   */
-  public async connectedCallback() {
-    super.connectedCallback()
-    this.provider =
-      typeof web3 !== 'undefined'
-        ? window.web3.currentProvider
-        : Web3.providers.HttpProvider('https://mainnet.infura.io')
-
-    this.web3 = new Web3(this.provider)
-    this.seaport = new OpenSeaPort(this.provider, {networkName: this.getNetwork()})
-    await this.connectWallet()
-    this.asset = await this.seaport.api.getAsset(this.contractAddress, this.tokenId);
-
-    [this.account] = await this.web3.eth.getAccounts()
-
-    // TODO: Check if account exists
-    if(this.account) {
-        this.asset.isOwnedByAccount = this.asset.owner.address.toLowerCase() === this.account.toLowerCase()
+
+    /**
+     * ConnectedCallback - Invoked when a component is added to the document’s DOM.
+     * Grabs data from the OpenSea SDK and populates data objects to be passed to
+     * child components.
+     */
+    public async connectedCallback() {
+        super.connectedCallback()
+
+        this.seaport = new OpenSeaPort(this.provider, {networkName: this.getNetwork()})
+        await this.connectWallet()
+
+        ;[this.account] = await this.web3.eth.getAccounts()
+
+        await this.getAsset(this.contractAddress, this.tokenId)
+
+        if (this.account) {
+            this.asset.isOwnedByAccount = this.asset.owner.address.toLowerCase() === this.account.toLowerCase()
```

Comment:
this looks like it should be a special getter method on this.asset. We've run into lots of bugs adding special properties onto assets in the frontend - let's instead make this kind of thing a computed property, so that if `this.asset.owner` changes for some reason, the value of `isOwnedByAccount` isn't stale

Reply:
so with current implementation `this.asset.owner` won't update unless page is refreshed. Right now component can not respond to address changes. But once it does I'll add logic to update the `isOwnedByAccount` &mdash; sound good?

Reply:
@alexanderatallah